### PR TITLE
feat(framework): add sanitize html to step output

### PR DESCRIPTION
**Pull request by davidsoderberg**
**What?**
* Sanitize the step output for all channel steps to eliminate XSS attacks via Studio and downstream content renderers such as Inbox.

**Screenshots**
_Step result is sanitized for email_
image

### CODE REVIEW

File: packages/framework/src/client.ts [hunk 1]
```diff
@@ -291,7 +292,15 @@ export class Client {
         setResult(stepResult);
       }
 
-      return stepResult.outputs as any;
+      if (
+        (
+          [ChannelStepEnum.CHAT, ChannelStepEnum.EMAIL, ChannelStepEnum.SMS, ChannelStepEnum.IN_APP] as string[]
```
Comment (rifont) [inline packages/framework/src/client.ts]: Need `PUSH` too

File: packages/framework/src/client.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ import type {
 } from './types';
 import { EMOJI, log } from './utils';
 import { transformSchema, validateData } from './validators';
+import { sanitizeHTML, sanitizeHtmlInObject } from './utils/sanitiez-html.utils';
```
Comment (rifont) [inline packages/framework/src/client.ts]: `sanitizeHTML` appears to be unused

File: packages/framework/src/utils/sanitiez-html.utils.ts
Comment (rifont) [inline packages/framework/src/utils/sanitiez-html.utils.ts:1]: I think we need to add some tests for:
1. unit test - verify that nested objects are sanitized
2. Integration test - verify that channel steps are sanitized, whilst other steps like digest/delay/custom are not sanitized

File: packages/framework/package.json [hunk 3]
```diff
@@ -132,25 +132,26 @@
...
+    "@apidevtools/json-schema-ref-parser": "11.6.4",
     "@sveltejs/kit": "^1.27.3",
+    "@types/aws-lambda": "^8.10.141",
     "@types/express": "^4.17.13",
     "@types/node": "^20.14.10",
-    "@types/aws-lambda": "^8.10.141",
     "@types/sanitize-html": "2.11.0",
     "@vercel/node": "^2.15.9",
+    "aws-lambda": "^1.0.7",
+    "eslint-plugin-unused-imports": "^4.0.0",
```
Comment (rifont) [inline packages/framework/package.json]: What if we moved this to the top-level package.json, disabled the rule by default, then just enabled the rule for this package (given that we extend the top-level eslint config). That way other packages can slowly opt-in and we keep a consistent eslint plugin version through the codebase.

File: packages/framework/src/client.ts [hunk 4]
```diff
@@ -35,6 +35,7 @@ import type {
 } from './types';
 import { EMOJI, log } from './utils';
 import { transformSchema, validateData } from './validators';
+import { sanitizeHtmlInObject } from './utils/sanitiez-html.utils';
```
Comment (rifont) [inline packages/framework/src/client.ts]: typo:
```suggestion
import { sanitizeHtmlInObject } from './utils/sanitize-html.utils';
```

File: packages/framework/src/client.ts [hunk 5]
```diff
@@ -291,7 +292,21 @@ export class Client {
         setResult(stepResult);
       }
 
-      return stepResult.outputs as any;
+      if (
```
Comment (SokratisVidros) [inline packages/framework/src/client.ts]: Instead of checking on specific channels, should we check if `stepResult.outputs` is typeof `string`?

**Review by SokratisVidros — APPROVED** [review]

File: packages/framework/src/client.ts [hunk 6]
```diff
@@ -291,7 +292,21 @@ export class Client {
...
+        (
+          [
+            ChannelStepEnum.CHAT,
+            ChannelStepEnum.EMAIL,
+            ChannelStepEnum.SMS,
+            ChannelStepEnum.IN_APP,
+            ChannelStepEnum.PUSH,
+          ] as string[]
+        ).includes(step.type)
```
Comment (rifont) [inline packages/framework/src/client.ts]:
```suggestion
        Object.values(ChannelStepEnum).includes(step.type as ChannelStepEnum)
```

File: packages/framework/src/client.ts (hunk 5 above)
Comment (davidsoderberg) [reply packages/framework/src/client.ts]: I think I like the check to be a bit more verbose so any future of us will easier understand what the check is for :)

**Pull request merged**